lib/active_storage_validations.rb

```diff
@@ -44,6 +44,50 @@ def content_type_valid?(file)
     end
 
   end
+  class ContentSizeValidator < ActiveModel::EachValidator
+    AVAILABLE_CHECKS = [:less_than, :less_than_or_equal_to, :greater_than, :greater_than_or_equal_to, :between]
+    
+    def check_validity!
+      unless (AVAILABLE_CHECKS).any? { |argument| options.has_key?(argument) }
+        raise ArgumentError, "You must pass either :less_than, :greater_than, or :between to the validator"
+      end
+    end
+    
+    def validate_each(record, attribute, value)
+      files = record.send(attribute)
+      # only attached
+      return true unless files.attached?
+
+      files = Array.wrap(files)
+      
+      files.each do |file|
+        if content_size_valid?(file)
+          record.errors.add(attribute, options[:message].presence || "size #{human_size(file.blob.byte_size)} is not between rquired range" )
+          return
+        end
+      end
+    end
+
+    def content_size_valid?(file)
+      file_size = file.blob.byte_size
+      case
+        when options[:between].present?
+          options[:between].include?(file_size)
+        when options[:less_than].present?
+          file_size > options[:less_than]
+        when options[:less_than_or_equal_to].present?
+          file_size >= options[:less_than_or_equal_to]
+        when options[:greater_than].present?
+          file_size < options[:greater_than]
+        when options[:greater_than_or_equal_to].present?
+          file_size <= options[:greater_than_or_equal_to]
+      end 
+    end
+
+    def human_size(size)
+      ActiveSupport::NumberHelper.number_to_human_size(size)
```

Comment:
I think you can do something like

delegate :number_to_human_size, to: ActiveSupport::NumberHelper

and remove this method